src/PrestaShopBundle/ApiPlatform/Provider/QueryProvider.php

```diff
@@ -32,51 +32,49 @@
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
-use PrestaShopBundle\ApiPlatform\Converters\ConverterInterface;
 use PrestaShopBundle\ApiPlatform\Exception\NoExtraPropertiesFoundException;
+use PrestaShopBundle\ApiPlatform\DomainSerializer;
+use ReflectionException;
 use Symfony\Component\Serializer\Exception\ExceptionInterface;
-use Symfony\Component\Serializer\Serializer;
 
 class QueryProvider implements ProviderInterface
 {
+    /**
+     * @param CommandBusInterface $queryBus
+     * @param DomainSerializer $apiPlatformSerializer
+     */
     public function __construct(
         private readonly CommandBusInterface $queryBus,
-        private readonly iterable $converters,
-        private readonly Serializer $apiPlatformSerializer
+        private readonly DomainSerializer    $apiPlatformSerializer,
     ) {
     }
 
     /**
-     * @throws \ReflectionException
-     * @throws \Exception
+     * @param Operation $operation
+     * @param array $uriVariables
+     * @param array $context
+     *
+     * @return mixed
+     *
      * @throws ExceptionInterface
+     * @throws NoExtraPropertiesFoundException
+     * @throws ReflectionException
      */
-    public function provide(Operation $operation, array $uriVariables = [], array $context = [])
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): mixed
     {
         $queryClass = $operation->getExtraProperties()['query'] ?? null;
         $filters = $context['filters'] ?? [];
         $queryParameters = array_merge($uriVariables, $filters);
```

Comment:
We'll probably have to handle the merge of these parameters in the command processor later as well no?